api/src/indigo_structure_checker.cpp

```diff
@@ -0,0 +1,205 @@
+/****************************************************************************
+ * Copyright (C) from 2009 to Present EPAM Systems.
+ *
+ * This file is part of Indigo toolkit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+#include "indigo_structure_checker.h"
+#include "api/indigo.h"
+#include "api/src/indigo_molecule.h"
+#include "api/src/indigo_reaction.h"
+#include "base_cpp/tlscont.h"
+#include "molecule/molecule_automorphism_search.h"
+#include "third_party/rapidjson/stringbuffer.h"
+#include "third_party/rapidjson/writer.h"
+#include <functional>
+#include <regex>
+#include <string>
+
+using namespace indigo;
+
+struct CheckParams
+{
+    int check_flags = StructureChecker2::CHECK_ALL;
+    std::vector<int> selected_atoms;
+    std::vector<int> selected_bonds;
+};
+
+static constexpr const char* checkTypeName[] = {
```

Comment:
code duplicate. It is worth to keep all the check types in structure_checker2 (see comment below)

Reply:
This is not code duplicate.
structure_checker2 works with bitwise values only and knows nothing about text values. 
text values are used by indigo_structure_checker only; structure_checker2 doesn't parse them 

So, it's not logical to define text values in structure_checker2 - they are related to API level and are not known in Core.